src/components/tasks/card/SuggestionBox.tsx

```diff
@@ -0,0 +1,107 @@
+import { FC, useEffect, useState } from 'react';
+import { DUMMY_PROFILE as placeholderImageURL } from '@/constants/display-sections';
+import classes from '@/components/tasks/card/suggestion.module.scss';
+import fetch from '@/helperFunctions/fetch';
+import { userDataType } from '@/interfaces/user.type';
+
+type Props = {
+    assigneeName: string;
+    showSuggestion: boolean;
+    onClickName: (userName: string) => void;
+};
+
+type GithubInfo = {
+    github_id: string;
+    url: string;
+};
+
+const SuggestionBox: FC<Props> = ({
+    assigneeName,
+    showSuggestion = false,
+    onClickName = () => undefined,
+}) => {
+    const [loading, setLoading] = useState<boolean>(false);
+    const [suggestions, setSuggestions] = useState<GithubInfo[]>([]);
+
+    const clickHandler = (e: any) => {
+        onClickName(e.target.innerText);
+    };
+
+    const fetchUsers = async () => {
+        setLoading(true);
+        const url = `${process.env.NEXT_PUBLIC_BASE_URL}/users?search=${assigneeName}`;
+        const { requestPromise } = fetch({ url });
+        const users = await requestPromise;
+        const usersData = users.data.users;
+        const suggestedUsers: GithubInfo[] = [];
+        usersData.map((data: userDataType) => {
+            suggestedUsers.push({
+                github_id: data.github_id,
+                url: data.picture
+                    ? data.picture.url
+                        ? data.picture.url
+                        : placeholderImageURL
+                    : placeholderImageURL,
```

Comment:
please try to optimize this, it's confusing me

Reply:
Sure i wll refactor this.
In first condition we are just checking for if picture property exists or not. And in the nested condition we check if url property exists or not on picture object.
In both cases if it gives false, we use placeholderImageURL

Reply:
resolved @Pratiyushkumar